docs/src/constraints.md

```diff
@@ -235,6 +235,59 @@ The constraint can also be retrieved from its name using
 constraint_by_name
 ```
 
+## Start Values
+
+Provide a starting value (also called warmstart) for a constraint's dual using
+[`set_dual_start_value`](@ref).
+
+The start value of a constraint's dual can be queried using [`dual_start_value`](@ref).
+If no start value has been set, [`dual_start_value`](@ref) will return `nothing`.
+
+```jldoctest constraint_dual_start; setup=:(model=Model())
+julia> @variable(model, x)
+x
+
+julia> @constraint(model, con, x >= 10)
+con : x ≥ 10.0
+
+julia> dual_start_value(con)
+
+julia> set_dual_start_value(con, 2)
+
+julia> dual_start_value(con)
+2.0
+```
+
+A vector constraint will require a vector warmstart:
+
+```jldoctest constraint_dual_start_vector; setup=:(model=Model())
+julia> @variable(model, x)
+x
+
+julia> @constraint(model, con, [x, x] in SecondOrderCone())
+con : [x, x] in MathOptInterface.SecondOrderCone(2)
+
+julia> dual_start_value(con)
+
+julia> set_dual_start_value(con, [1.0, 3.0])
+
+julia> dual_start_value(con)
+2-element Array{Float64,1}:
+ 1.0
+ 3.0
+```
+
+To take the dual solution from the last solve and use it as the starting point
+for a new solve, use:
+
+```julia
+for (F, S) in list_of_constraint_types(model)
+    for con in all_constraints(model, F, S)
+        set_dual_start_value(con, dual(con))
+    end
+end
+```
```

Comment:
Note that this will error if there are constraints for which the dual is not defined. You may want to choose only a subset of `F`-`S` pairs to set a dual start for.

Reply:
This might be a little complicated...
You want to add filters in the example or just a note?

Reply:
I was proposing adding the sentence as a note.

Reply:
done! let me know if its ok